subprojects/workers/src/main/java/org/gradle/workers/internal/ActionExecutionSpec.java

```diff
@@ -32,15 +32,15 @@
 public class ActionExecutionSpec implements WorkSpec {
     private final String displayName;
     private final Class<? extends Runnable> implementationClass;
-    private final File defaultDir;
-    private final File workingDir;
+    private final File idleWorkingDir;
```

Comment:
Well, I agree this isn't the best.

The problem is that `WorkerDaemonServer` is instantiated by the `WorkerProcess` infrastructure which relies on the server implementations to have no argument constructors.  Just sending this value with the spec seemed to be the simplest, least-worst solution.  

I've changed this to use a dependency injecting instantiator instead and inject an instance of `WorkerDirectoryProvider` on the worker process side (which is probably what I should have done here in the first place).